### DISCUSSION TITLE
Slow responses from GraphQL in production hosted on vercel

### DISCUSSION THREAD

**Discussion by neilkuumar** [vercel/vercel#6933] [Help]
I have a GraphQL resolver that does a really simple fetch to google firestore and returns the results.  

On my local machine this takes less than 100ms, however, when deployed on vercel this can take between 600ms to 5 seconds to return the data.  

I know the default vercel server location is in Washington and my Firebase default server location is in the EU. Could this really be the issue here?  Is there some optimisation I need to do when deploying GraphQL to vercel?

**Comment by leerob** [comment]
Yes, that's definitely a part of the issue. Your functions should be close to your database (e.g. Firestore). I would recommend changing your function location to EU, or moving the Firebase location to `us-east`.

**Reply by neilkuumar** [reply]
Thanks for response! The firestore location cannot be changed after creation unfortunately and to be able to change the vercel location I need to pay for the pro plan right?
